src/badge_manager_standalone.py

```diff
@@ -36,22 +36,42 @@ def _read_file(self,device_file):
         self.logger.info("Reading devices from file: {}".format(device_file))
 
         regex = re.compile(r'^([A-Fa-f0-9]{2}(?::[A-Fa-f0-9]{2}){5}).*')
+       
+        #extracting badge id and project from device file
+        with open(device_file, 'r') as devices_macs:
+            ids = [re.split(" +",lines)[2:] for lines in devices_macs]
+            badge_project_ids = ids[2:]
+            badge_ids = [arr[0] for arr in badge_project_ids]
+            project_ids = [arr[1] for arr in badge_project_ids]
+            
 
+        #extracting mac addresses of the badge    
         with open(device_file, 'r') as devices_macs:
-            devices = [regex.findall(line) for line in devices_macs]
+
+            devices = [regex.findall(line) for line in devices_macs]                                        
             devices = filter(lambda x: x, map(lambda x: x[0] if x else False, devices))
             devices = [d.upper() for d in devices]
+        
 
+        #mapping badge id and project id to mac address
+        mac_id_map = {}    
+        for i in range(len(devices)):
+            mac_id_map[devices[i]] = badge_project_ids[i]
+
+            
+        
         for d in devices:
             self.logger.debug("    {}".format(d))
 
         badges = {mac: Badge(mac,
                                        self.logger,
-                                       key=mac,  # using mac as key since no other key exists
+                                       key=mac,  # using mac as key since no other key exists or can we use badge id or project for it ??
```

Comment:
Use the mac here. The Key is a "person" identifier that we get from the server in server mode. It allows us to give a new badge to a person (different mac) but know it's the same person. Please remove the addition to the comment you put here.